### ISSUE TITLE
Documentation Request: Include subsection links in documentation sidebar

### ISSUE THREAD

**Issue by DanKaplanSES** [jqlang/jq#3031]
For context, I was trying to learn the difference between `=` and `|=`. The sidebar includes [Assignment](https://jqlang.github.io/jq/manual/v1.7/#assignment) but it does not include links to the subsection named [Update assignment: `|=`](https://jqlang.github.io/jq/manual/v1.7/#update-assignment) or [Plain assignment: `=`](https://jqlang.github.io/jq/manual/v1.7/#plain-assignment). Because another section divides these two, I had to scroll back and forth to read, and often went clicked on the wrong place by clicking on [Advanced Features](https://jqlang.github.io/jq/manual/v1.7/#advanced-features) instead of "Assignment."

To put it another way, I would like another level of nesting for each section so, for example, there would be links for "Update assignment" and "Plain assignment" at this red line:

![image](https://github.com/jqlang/jq/assets/373675/96eebd91-071e-49d2-a185-ee2e2b916374)

**Environment (please complete the following information):**

- OS and Version: [e.g. macOS, Windows, Linux (please specify distro)]: Windows 10, WSL Ubuntu
- jq version [e.g. 1.5]: 1.6

**Comment by wader** [comment]
Had a quick look at adding a subsection for each entry for a section in the content tree. Ran into two issues:
- Some section has quite a lot of entries, might make the table of contents quite long. Maybe could be solved by folding somehow? see below
- Didn't find any clear way with bootstrap to generate a "tree" table of contents that still shows the "active" section when scrolling. Someone who knows bootstrap maybe knows how?

A possible other solution might be to move out some entries to become sections instead?
